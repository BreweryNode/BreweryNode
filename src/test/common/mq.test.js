jest.genMockFromModule('amqplib');
const mq = require('../../main/common').mq;

describe('mq', () => {
  it('fails_to_connect', () => {
    expect.assertions(1);
    return expect(
      mq.connect('amqp://localhost:1', 'amq.topic')
    ).rejects.toThrowErrorMatchingSnapshot();
  });

  it('connects', () => {
    expect.assertions(1);
    return expect(
      mq.connect(
        'amqp://jgvoxqek:b_W1ueA3zx7l8YSYFLGzfJ4qFPEyxBDR@penguin.rmq.cloudamqp.com/jgvoxqek',
        'amq.topic'
      )
    ).resolves.toBeDefined();
  });

  let message;

  function ensureMessage() {
    return new Promise(function(resolve) {
      (function waitForFoo() {
        if (message) {
          return resolve(message.content.toString());
        }
        setTimeout(waitForFoo, 30);
      })();
    });
  }

  it('subscribes', async () => {
    expect.assertions(1);
    return expect(
      mq.recv('test', 'test.v1.#', rec => {
        message = rec;
      })
    ).resolves.toBeDefined();
  });

  it('sends', async () => {
    expect.assertions(1);
    return expect(mq.send('test.v1.testmq', 'This is a test')).resolves.toBeUndefined();
  });

  it('receives', async () => {
    expect.assertions(1);
    return expect(ensureMessage()).resolves.toMatchSnapshot();
  });

  it('disconnects', () => {
    expect.assertions(1);
    return expect(mq.disconnect()).resolves.toBeUndefined();
  });

  it('fails_to_disconnect', () => {
    expect.assertions(1);
    return expect(mq.disconnect()).rejects.toThrowErrorMatchingSnapshot();
  });

  it('doesnt-send', async () => {
    expect.assertions(1);
    return expect(
      mq.send('test.v1.testmq', 'This is a test')
    ).rejects.toThrowErrorMatchingSnapshot();
  });

  it('doesnt-send', async () => {
    expect.assertions(1);
    return expect(mq.send('test.v1.testmq')).rejects.toThrowErrorMatchingSnapshot();
  });
});
