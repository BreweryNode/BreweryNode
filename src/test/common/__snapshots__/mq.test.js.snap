// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mq doesnt-send 1`] = `"Channel closed"`;

exports[`mq fails_to_connect 1`] = `"connect ECONNREFUSED 127.0.0.1:1"`;

exports[`mq fails_to_disconnect 1`] = `"Connection closed (by client)"`;

exports[`mq receives 1`] = `"This is a test"`;
